netlogo-gui/src/main/lab/gui/Supervisor.scala

```diff
@@ -89,6 +90,12 @@ class Supervisor(dialog: java.awt.Dialog,
         worker.protocol,
         new java.io.PrintWriter(new java.io.FileWriter(fileName))))
     }
+    if(options.updateView){
+      progressDialog.setUpdateView(true)
+    }
+    if(options.updatePlotsAndMonitors){
+      progressDialog.setPlotsAndMonitorsSwitch(true)
+    }
```

Comment:
yes , when checkbox is selected in RunOptions dialog this option is set to true, so when the option is set to true corresponding method in progressdialog is also set to true. Thus changes are made respectively.

Reply:
My question: is there a reason to write it this way instead of just `progressDialog.setPlotsAndMonitorsSwitch(options.updatePlotsAndMonitors)`?

Reply:
i have not tried it the other way, but the way you mentioned does the same thing. Instead of checking the boolean value we can directly pass as its function argument as you mentioned.